File: src/main/java/com/IconPippi/simtriggers/wrappers/TextLine.java
package com.IconPippi.simtriggers.wrappers;

import java.io.IOException;

import com.IconPippi.simtriggers.events.EventFactory;
import com.IconPippi.simtriggers.utils.Logger;

import flightsim.simconnect.SimConnect;
import flightsim.simconnect.TextType;

/**
 * This class simplifies the text line generating process
 * @author IconPippi
 *
 */
public class TextLine {
	
	/* SimTriggers constant */
	private final SimConnect sc = SimTriggers.getSimulator();
	
	/* Logger */
	private final Logger logger = new Logger();
	
	/*
	 * Text line data
	 */
	private String text;
	private int timeout;
	private TextType textColor;
	private int textLineID;
	
	/**
	 * Initialize a text line
	 */
	public TextLine() {
		textLineID = new EventFactory().buildTextLine();
	}
	
	/**
	 * Sets the text line's text 
	 * @param text text line's text
	 */
	public void setText(String text) {
		this.text = text;
	}
	
	/**
	 * Sets the text line's seconds timeout
	 * @param timeout text line's timeout
	 */
	public void setTimeout(int timeout) {
		this.timeout = timeout;
	}
	
	/**
	 * Sets the text color
	 * @see {@link flightsim.simconnect.TextType} 
	 * @param color text's color
	 */
	public void setTextColor(TextType color) {
		this.textColor = color;
	}
	
	/**
	 * Show the text line
	 */
	public void show() {
		try {
			sc.text(textColor, timeout, textLineID, text);
		} catch (UnsupportedOperationException | IOException e) {
			logger.error(e.toString());
		}
	}
}

File: src/main/kotlin/com/simtriggers/fsx/simulator/TextLine.kt
package com.simtriggers.fsx.simulator

import com.simtriggers.fsx.SimTriggers
import com.simtriggers.fsx.event.RegisterEvent
import flightsim.simconnect.SimConnect
import flightsim.simconnect.TextType

/**
 * 06/08/2019
 * FSX text line
 *
 * @author IconPippi
 */
class TextLine {

    /** SimTriggers constant */
    private val sc: SimConnect = SimTriggers.sc

    /** Event registerer constant */
    private val eventRegisterer: RegisterEvent = RegisterEvent()

    /** Text line identifier */
    private val textLineID: Int


    private var text: String? = null
    private var timeout: Float? = null
    private var textColor: TextType? = null

    /**
     * Create a new text line
     */
    init {
        textLineID = eventRegisterer.registerTextLineEvent()
    }

    /**
     * Set the text line's text
     * @param text Text line's text
     */
    fun setText(text: String) {
        this.text = text
    }

    /**
     * Set the text line's timeout
     * @param timeout Text line's timeout
     */
    fun setTimeout(timeout: Float) {
        this.timeout = timeout
    }

    /**
     * Set the text line's color
     * @param color Text line's color
     */
    fun setTextColor(color: TextType) {
        this.textColor = color
    }

    /**
     * Show the text line
     */
    fun show() {
        sc.text(textColor, timeout!!, textLineID, text)
    }

}